sklearn/ensemble/tests/test_forest.py

```diff
@@ -91,16 +91,16 @@
 FOREST_ESTIMATORS.update(FOREST_TRANSFORMERS)
 
 
-def check_classification_toy(name):
+def check_classification_toy(name, max_samples):
     """Check classification on a toy dataset."""
     ForestClassifier = FOREST_CLASSIFIERS[name]
 
-    clf = ForestClassifier(n_estimators=10, random_state=1)
+    clf = ForestClassifier(n_estimators=10, random_state=1, max_samples=max_samples)
     clf.fit(X, y)
     assert_array_equal(clf.predict(T), true_result)
     assert_equal(10, len(clf))
 
-    clf = ForestClassifier(n_estimators=10, max_features=1, random_state=1)
+    clf = ForestClassifier(n_estimators=10, max_features=1, random_state=1, max_samples=max_samples)
```

Comment:
My PR has tests for the changes in line with existing tests in the class. You suggest a different work on improving the test cases. My interest in this PR is in bringing Random Forests in line with Breiman (2001) by sampling training instances. This is very useful when training over large sets. I have used this code for my own experiments and it works, I hope others might benefit from this improvement which puts scikit-learn implementation in line with other ML libraries and published work on RFs.

I offer improving the test cases of the class as a compromise, seeing that is something that interests you.